base/src/main/java/proguard/optimize/inline/BaseLambdaInliner.java

```
@@ -0,0 +1,357 @@
+        ) {
+            partialEvaluator.visitCodeAttribute(consumingClass, consumingMethod, consumingMethodCodeAttribute);
+            TracedStack tracedStack = partialEvaluator.getStackBefore(consumingMethodCallOffset);
+
+            // If we have a lambda with n arguments we have to skip those n arguments and then get the instance of the lambda used for calling invokeinterface.
+            int lambdaInstanceStackIndex = ClassUtil.internalMethodParameterCount(invokeInterfaceMethod.getDescriptor(invokeInterfaceClass));
+
+            consumingMethodCodeAttribute.instructionAccept(
+                consumingClass,
+                consumingMethod,
+                tracedStack.getTopActualProducerValue(lambdaInstanceStackIndex).instructionOffsetValue().instructionOffset(0),
+                new LambdaInvokeUsageReplacer(consumingMethodCallOffset)
+            );
+        }
+    }
+
+    @Override
+    public void visitInterfaceMethodrefConstant(Clazz clazz, InterfaceMethodrefConstant interfaceMethodrefConstant) {
+        referencedInterfaceConstant = interfaceMethodrefConstant;
+    }
+
+    private class LambdaInvokeUsageReplacer implements InstructionVisitor {
+        private final int possibleLambdaInvokeCallOffset;
+
+        public LambdaInvokeUsageReplacer(int possibleLambdaInvokeCallOffset) {
+            this.possibleLambdaInvokeCallOffset = possibleLambdaInvokeCallOffset;
+        }
+
+        @Override
+        public void visitVariableInstruction(Clazz clazz, Method method, CodeAttribute codeAttribute, int offset, VariableInstruction variableInstruction) {
+            // Uses same codeAttributeEditor as the casting remover
+            Instruction tracedInstruction = SourceTracer.traceParameter(partialEvaluator, codeAttribute, offset);
+            if (tracedInstruction.canonicalOpcode() == Instruction.OP_ALOAD) {
+                variableInstruction = (VariableInstruction) tracedInstruction;
```

What if the tracedInstruction is not a `VariableInstruction`? Then we should throw an exception?